### PR TITLE
improve useAtom error message

### PR DESCRIPTION
**Pull request by Aslemammad**
@dai-shi I was playing with codesandbox, then jotai showed me that error, but I had a provider, then I realized that I didn't put the scope. Maybe helpful for the folks.

### CODE REVIEW

File: src/core/useAtom.ts [hunk 1]
```diff
@@ -6,7 +6,9 @@ import { Atom, WritableAtom, AnyWritableAtom } from './types'
 
 function assertContextValue<T extends object>(x: T | null): asserts x is T {
```
Comment (dai-shi) [inline src/core/useAtom.ts:7]:
```suggestion
function assertContextValue<T extends object>(x: T | null, scope?: unknown): asserts x is T {
```

Let's show the note about the scope only if scope is specified. I suppose, the majority of users don't use scopes.

**Review by dai-shi — CHANGES_REQUESTED** [review]
Thanks for the suggestion! can you change a bit?

File: src/core/useAtom.ts [hunk 2]
```diff
@@ -6,7 +6,9 @@ import { Atom, WritableAtom, AnyWritableAtom } from './types'
...
   if (!x) {
-    throw new Error('Please use <Provider>')
+    throw new Error(
+      'Please use <Provider> or if you have scoped atoms, use <Provder scope={yourAtomScope}>'
```
Comment (dominictwlee) [inline src/core/useAtom.ts:10]:
```suggestion
      'Please use <Provider> or if you have scoped atoms, use <Provider scope={yourAtomScope}>'
```

**Comment by Aslemammad** [comment]
I think it's ready now.

**Review by dai-shi — APPROVED** [review]
LGTM

**Pull request merged**